### PR TITLE
main.c changes

### PR DESCRIPTION
**Pull request by rdentato**
read_file() exited without closing the file. Now it does, regardless the error status.
Slightly simplified file_exists().

### CODE REVIEW

**Review by foxweb — COMMENTED** [review]
My skills in C are beginner.
Why do you think that transfer defines from main.c is a good practice?

File: main.c [hunk 1]
```diff
@@ -32,33 +11,29 @@ int main(int c, char **v) {
 
 int file_exists(const char *file_name) {
   struct stat buffer;
-  int exist = stat(file_name, &buffer);
-  if (exist == 0)
-    return 1;
-  else
-    return 0;
+  int exist;
```
Comment (foxweb) [inline main.c:14]: exist**s**

**Comment by rdentato** [comment]
> Why do you think that transfer defines from main.c is a good practice?

I see httpd.c mainly as a library it could be used to easily create REST servers in C (which is my intent here). If that's the case, all the defines and declarations should be in the http.h header so that one can link httpd.c to different "main" for different purpose.

File: main.c (hunk 1 above)
Comment (rdentato) [reply main.c:14]: Sure. Do you want me to change it?

**Comment by rdentato** [comment]
I'll close this pull request and open a new one.

**Comment by foxweb** [comment]
@rdentato you don't need to create a new pull request every time! make changes within the same branch!

**Pull request closed without merging**